recipes/jasper/all/conanfile.py

```diff
@@ -38,27 +42,30 @@ def _build_subfolder(self):
 
     def export_sources(self):
         self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
     def requirements(self):
         if self.options.with_libjpeg == "libjpeg-turbo":
             self.requires("libjpeg-turbo/2.1.2")
         elif self.options.with_libjpeg == "libjpeg":
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "16":
```

Comment:
```suggestion
        if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) == "16":
```

I disagree with this strategy, but at least please don't raise for Visual Studio 2022.